### PR TITLE
feat: [SRM-12485]: Add Fail on NO Analysis

### PR DESCRIPTION
**Pull request by vasuki-kalluri**
#### Screenshots

NA
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/35-connectors/strings/strings.en.yaml [hunk 1]
```diff
@@ -391,6 +391,7 @@ cdng:
   baseline: Baseline
   trafficsplit: Traffic Split (OPTIONAL)
   continousVerificationStep: Verify Step Details
+  failOnNoAnalysis: Fail On No Analysis
```
Comment (praneshg239) [inline src/modules/35-connectors/strings/strings.en.yaml:394]: please check whether `Fail On No Analysis` or `Fail on No Analysis`. Casing

**Review by praneshg239 — DISMISSED** [review]

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx [hunk 2]
```diff
@@ -123,6 +123,9 @@ export default function ConfigureFields(props: {
           multiTextInputProps={{ expressions, allowableTypes }}
         />
       </div>
+      <div className={cx(stepCss.formGroup)}>
```
Comment (avinashmadhwani02) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx:126]: why do we need cx if there's only on class

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx [hunk 3]
```diff
@@ -123,6 +123,9 @@ export default function ConfigureFields(props: {
...
+        <FormInput.CheckBox name="spec.spec.failOnNoAnalysis" label={getString('connectors.cdng.failOnNoAnalysis')} />
```
Comment (harshilbits) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx:127]: Have we tested this ? Is the failOnAnalysis value getting saved in the pipeline yaml once we save the pipeline ?

Comment (vasuki-kalluri) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx:126]: Yes. It gets saved in yaml

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx (hunk 2 above)
Comment (vasuki-kalluri) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationWidget/components/ContinousVerificationWidgetSections/components/ConfigureFields/ConfigureFields.tsx]: removed it

**Review by praneshg239 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Comment by vasuki-kalluri** [comment]
trigger cypress

**Comment by vasuki-kalluri** [comment]
trigger cypress

**Comment by vasuki-kalluri** [comment]
trigger cypress

**Pull request merged**